torch/quantization/observer.py

```diff
@@ -682,50 +681,44 @@ def _compute_quantization_error(next_start_bin, next_end_bin, norm_type):
             """
             bin_width = (self.max_val.item() - self.min_val.item()) / self.bins
 
-            norm = 0.0
             dst_bin_width = bin_width * (next_end_bin - next_start_bin + 1) / self.dst_nbins
             if dst_bin_width == 0.0:
                 return 0.0
-            for src_bin in range(self.bins):
-                # distances from the beginning of first dst_bin to the beginning and
-                # end of src_bin
-                src_bin_begin = (src_bin - next_start_bin) * bin_width
-                src_bin_end = src_bin_begin + bin_width
-
-                # which dst_bins the beginning and end of src_bin belong to?
-                dst_bin_of_begin = min(
-                    self.dst_nbins - 1, max(0.0, math.floor(src_bin_begin / dst_bin_width))
-                )
-                dst_bin_of_end = min(
-                    self.dst_nbins - 1, max(0.0, math.floor(src_bin_end / dst_bin_width))
-                )
-                dst_bin_of_begin_center = (
-                    dst_bin_of_begin * dst_bin_width + dst_bin_width / 2
-                )
-
-                density = self.histogram[src_bin] / bin_width
-                if dst_bin_of_begin == dst_bin_of_end:
-                    # if src_bin is entirely within 1 dst_bin
-                    delta_begin = src_bin_begin - dst_bin_of_begin_center
-                    delta_end = src_bin_end - dst_bin_of_begin_center
-                    norm = norm + _get_norm(delta_begin, delta_end, density, norm_type)
-                else:
-                    delta_begin = src_bin_begin - dst_bin_of_begin_center
-                    delta_end = dst_bin_width / 2
-                    norm = norm + _get_norm(delta_begin, delta_end, density, norm_type)
-
-                    norm = norm + (dst_bin_of_end - dst_bin_of_begin - 1) * _get_norm(
-                        -dst_bin_width / 2, dst_bin_width / 2, density, norm_type
-                    )
-
-                    dst_bin_of_end_center = (
-                        dst_bin_of_end * dst_bin_width + dst_bin_width / 2
-                    )
-
-                    delta_begin = -dst_bin_width / 2
-                    delta_end = src_bin_end - dst_bin_of_end_center
-                    norm = norm + _get_norm(delta_begin, delta_end, density, norm_type)
-            return norm
+
+            src_bin = torch.arange(self.bins).numpy()
```

Comment:
PyTorch doesn't have a NumPy dependency for its functionality (although we do for some tests), and we shouldn't use NumPy functionality in lieu of our own. Uses of NumPy should be restricted to testing and NumPy interop.

Reply:
Thanks for the feedback -- I changed my code to get rid of the numpy dependency.